core/src/main/java/io/kestra/core/repositories/LogRepositoryInterface.java

```diff
@@ -5,17 +5,23 @@
 import io.micronaut.data.model.Pageable;
 import org.slf4j.event.Level;
 
+import javax.annotation.Nullable;
 import java.time.ZonedDateTime;
 import java.util.List;
-import javax.annotation.Nullable;
 
 public interface LogRepositoryInterface extends SaveRepositoryInterface<LogEntry> {
     List<LogEntry> findByExecutionId(String id, Level minLevel);
 
+    ArrayListTotal<LogEntry> findByExecutionId(String id, Level minLevel, Pageable pageable);
+
     List<LogEntry> findByExecutionIdAndTaskId(String executionId, String taskId, Level minLevel);
 
+    ArrayListTotal<LogEntry> findByExecutionIdAndTaskId(String executionId, String taskId, Level minLevel, Pageable pageable);
+
     List<LogEntry> findByExecutionIdAndTaskRunId(String executionId, String taskRunId, Level minLevel);
 
+    ArrayListTotal<LogEntry> findByExecutionIdAndTaskRunId(String executionId, String taskRunId, Level minLevel, Pageable pageable);
+
     ArrayListTotal<LogEntry> find(
```

Comment:
Are the non-pageable methods still used ?

Reply:
They are used for the download methods